internal/gatewayapi/route.go

```diff
@@ -1158,12 +1136,12 @@ func (t *Translator) processDestination(backendRefContext BackendRefContext,
 
 	backendNamespace := NamespaceDerefOr(backendRef.Namespace, route.GetNamespace())
 	if !t.validateBackendRef(backendRefContext, parentRef, route, resources, backendNamespace, routeType) {
-		return nil, weight
+		return &ir.DestinationSetting{Weight: &weight}
```

Comment:
can we add a comment here highlighting that this is referring to an invalid weight